File: frontend/src/api/docs.ts
import { api } from "./client";

export interface Doc {
  id: number;
  type: string;
  category: string;
  title: string;
  description: string;
  url: string;
  content: string;
  thumbnail_url: string;
  duration_label: string;
  tags: string;
  sort_order: number;
}

export interface PublicDocsResult {
  categories: Record<string, Doc[]>;
}

export interface HelpSubmitPayload {
  type?: string;
  category?: string;
  title: string;
  description?: string;
  email?: string;
}

export function listDocs() {
  return api.get<PublicDocsResult>("/docs").then((r) => r.data);
}

export function submitHelpRequest(payload: HelpSubmitPayload) {
  return api.post<{ ok: boolean }>("/help", payload).then((r) => r.data);
}

File: src/presentation/schemas/docs.py
from apiflask import Schema
from apiflask.fields import Boolean, Integer, String


class HelpSubmitInSchema(Schema):
    type = String(load_default="")
    category = String(load_default="")
    title = String(required=True)
    description = String(load_default="")
    email = String(load_default="")


class DocInSchema(Schema):
    type = String(load_default="video")
    category = String(load_default="General")
    title = String(load_default="")
    description = String(load_default="")
    url = String(load_default="")
    content = String(load_default="")
    thumbnail_url = String(load_default="")
    duration_label = String(load_default="")
    tags = String(load_default="")
    sort_order = Integer(load_default=0)
    is_published = Boolean(load_default=True)
